### PR TITLE
feat: table of content improvements

### PR DESCRIPTION
**Pull request by miedziak**
- make table of content interactive while scrolling
- add second level of content based on h3

### CODE REVIEW

File: src/components/learn/TableOfContent/TableOfContent.js [hunk 1]
```diff
@@ -59,17 +60,33 @@ Item.propTypes = {
   isActive: PropTypes.bool.isRequired,
 };
 
-const TableOfContent = ({ items, activeHash }) => {
-  if(isEmptyTableOfContent(items)) return null;
+const getHashesFromItems = (items) => {
+  return items.reduce((acc, lvl1Item) => {
```
Comment (LWronka) [inline src/components/learn/TableOfContent/TableOfContent.js:64]: Q: May be lets check if is Array ?

File: src/hooks/useActiveHashInViewport.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { useScrollPosition } from '@n8tb1t/use-scroll-position';
+import {useState} from "react";
+
+const useActiveHashInViewport = (hashes = []) => {
+  const [activeViewportHash, setActiveViewportHash] = useState(null);
```
Comment (LWronka) [inline src/hooks/useActiveHashInViewport.js:5]: `null` is as default for `useState`

File: src/hooks/useActiveHashInViewport.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  useScrollPosition(() => {
+    if(hashes.length === 0) {
+      setActiveViewportHash(null);
+      return;
```
Comment (LWronka) [inline src/hooks/useActiveHashInViewport.js:9]: Q: is it `return` mandatory here ?

Comment (miedziak) [reply src/hooks/useActiveHashInViewport.js:10]: yes,

**Pull request merged**